lib/openstack_query/queries/query_server.py

```diff
@@ -0,0 +1,124 @@
+from enums.query.server_properties import ServerProperties
+from enums.query.query_presets import (
+    QueryPresetsGeneric,
+    QueryPresetsDateTime,
+    QueryPresetsString,
+)
+
+from openstack_query.queries.query_wrapper import QueryWrapper
+from openstack_query.runners.server_runner import ServerRunner
+
+from openstack_query.handlers.client_side_handler_generic import (
+    ClientSideHandlerGeneric,
+)
+from openstack_query.handlers.client_side_handler_string import ClientSideHandlerString
+from openstack_query.handlers.client_side_handler_datetime import (
+    ClientSideHandlerDateTime,
+)
+
+from openstack_query.handlers.server_side_handler import ServerSideHandler
+from openstack_query.handlers.prop_handler import PropHandler
+
+from openstack_query.utils import convert_to_timestamp
+
+
+class QueryServer(QueryWrapper):
+    """
+    Query class for querying Openstack Server objects.
+    Define property mappings, kwarg mappings and filter function mappings related to servers here
+    """
+
+    # PROPERTY_MAPPINGS set how to get properties of a openstack server object
+    # possible properties are documented here:
+    # https://docs.openstack.org/openstacksdk/latest/user/resources/compute/v2/server.html#openstack.compute.v2.server.Server
+    _PROPERTY_MAPPINGS = {
+        ServerProperties.USER_ID: lambda a: a["user_id"],
+        ServerProperties.HYPERVISOR_ID: lambda a: a["host_id"],
+        ServerProperties.SERVER_ID: lambda a: a["id"],
+        ServerProperties.SERVER_NAME: lambda a: a["name"],
+        ServerProperties.SERVER_DESCRIPTION: lambda a: a["description"],
+        ServerProperties.SERVER_STATUS: lambda a: a["status"],
+        ServerProperties.SERVER_CREATION_DATE: lambda a: a["created_at"],
+        ServerProperties.SERVER_LAST_UPDATED_DATE: lambda a: a["updated_at"],
+        ServerProperties.FLAVOR_ID: lambda a: ["flavor_id"],
+        ServerProperties.IMAGE_ID: lambda a: ["image_id"],
+        ServerProperties.PROJECT_ID: lambda a: a["location"]["project"]["id"],
+    }
+
+    # SERVER_SIDE_FILTER_MAPPINGS - 'filter' keyword arguments that openstack command conn.compute.servers() takes
+    # - documented here https://docs.openstack.org/openstacksdk/latest/user/proxies/compute.html
+    # - all filters are documented here
+    # https://docs.openstack.org/api-ref/compute/?expanded=list-servers-detail#list-server-request
+    _SERVER_SIDE_FILTER_MAPPINGS = {
+        QueryPresetsGeneric.EQUAL_TO: {
+            ServerProperties.USER_ID: lambda **kwargs: {"user_id": kwargs["value"]},
```

Comment:
Is this requirement to have
`(value="some_user")` for singular scalar values documented anywhere?

Since I (as a user) might expect it to be positional in this case

Reply:
the `EQUAL_TO` preset - which is the filter we are setting here - takes one parameter - `value` - which is the value to compare against.
Yes I think it should be a positional argument